tests/geo/test_surface_rzfourier.py

```diff
@@ -522,33 +522,30 @@ def test_change_resolution(self):
         """
         for mpol in [1, 2]:
             for ntor in [0, 1]:
-                s = SurfaceRZFourier(mpol=mpol, ntor=ntor)
-                n = len(s.get_dofs())
-                s.set_dofs((np.random.rand(n) - 0.5) * 0.01)
-                s.set_rc(0, 0, 1.0)
-                s.set_rc(1, 0, 0.1)
-                s.set_zs(1, 0, 0.13)
-                v1 = s.volume()
-                a1 = s.area()
-
-                s.change_resolution(mpol+1, ntor)
-                s.recalculate = True
+                s_orig = SurfaceRZFourier(mpol=mpol, ntor=ntor)
+                n = len(s_orig.get_dofs())
+                s_orig.set_dofs((np.random.rand(n) - 0.5) * 0.01)
+                s_orig.set_rc(0, 0, 1.0)
+                s_orig.set_rc(1, 0, 0.1)
+                s_orig.set_zs(1, 0, 0.13)
+                v1 = s_orig.volume()
+                a1 = s_orig.area()
+
+                s = s_orig.change_resolution(mpol+1, ntor)
                 v2 = s.volume()
                 a2 = s.area()
                 self.assertAlmostEqual(v1, v2)
                 self.assertAlmostEqual(a1, a2)
 
-                s.change_resolution(mpol, ntor+1)
-                s.recalculate = True
-                v2 = s.volume()
-                a2 = s.area()
+                s2 = s.change_resolution(mpol, ntor+1)
+                v2 = s2.volume()
+                a2 = s2.area()
                 self.assertAlmostEqual(v1, v2)
                 self.assertAlmostEqual(a1, a2)
 
-                s.change_resolution(mpol+1, ntor+1)
-                s.recalculate = True
-                v2 = s.volume()
-                a2 = s.area()
+                s3 = s2.change_resolution(mpol+1, ntor+1)
+                v2 = s3.volume()
+                a2 = s3.area()
```

Comment:
Can you add a test here that checks if it fixes Issue #478? e.g. tests that s.darea_by_dcoeff() and other surface methods can be called after changing the resolution?